File: python-prov/src/checking_uri_syntax.py
from prov.model import ProvDocument

from test_case import TestCase


class CheckingUriSyntax(TestCase):
    def __init__(self):
        self.filename = "checking_uri_syntax"

    def create_document(self):
        document = ProvDocument()
        document.add_namespace("ex", "http://www.w3. org/ns/prov#")
        document.entity("ex:e")
        return document

File: java-prov/src/main/java/org/example/CheckingUriSyntax.java
package org.example;

import org.openprovenance.prov.interop.InteropFramework;
import org.openprovenance.prov.model.Entity;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.vanilla.Document;
import org.openprovenance.prov.vanilla.ProvFactory;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;


public class CheckingUriSyntax extends TestCase {

    public CheckingUriSyntax(){
        setFilename("checking_uri_syntax");
    }
    public Document createDocument(){
        var factory = new ProvFactory();
        var document = new Document();
        var ns = new Namespace();
        ns.addKnownNamespaces();
        ns.register("ex","http://www.w3. org/ns/prov#");
        var e = ns.qualifiedName("ex","e",factory);
        Entity entity = factory.newEntity(e);
        document.getStatementOrBundle().add(entity);
        document.setNamespace(ns);
        return document;
    }
}
